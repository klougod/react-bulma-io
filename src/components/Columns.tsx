import React from 'react'
import PropTypes from 'prop-types'

export const Columns = ({ className, children }: any) => (
  <div className={`columns ${className}`}>{children}</div>
)

Columns.defaultProps = {
  className: ''
}

Columns.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default Columns
