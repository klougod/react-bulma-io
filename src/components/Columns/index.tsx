import React, { ReactNode } from 'react'

interface ColumnsProps {
  className?: string
  children?: ReactNode
}

export const Columns = ({ className, children }: ColumnsProps) => (
  <div className={`columns ${className}`}>{children}</div>
)

Columns.defaultProps = {
  className: ''
}

export default Columns
