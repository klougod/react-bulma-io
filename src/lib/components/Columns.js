import React from 'react'
import PropTypes from 'prop-types'

export const Columns = ({ className, children }) => (
  <div className={`columns ${className}`}>
    {children}
  </div>
)

Columns.defaultProps = {
  className: ''
}

Columns.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default Columns