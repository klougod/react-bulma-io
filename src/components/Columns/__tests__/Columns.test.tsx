import React from 'react'
import Enzyme from 'enzyme'

import Columns from '../'

describe('Columns component test', function () {
  it('should render without crash without props', function () {
    const columnsMock = Enzyme.shallow(<Columns />)
    expect(columnsMock).toMatchSnapshot()
  })

  it('should render without crash with props', function () {
    const columnsMockProps = {
      className: 'columnsCLass',
    }
    const columnsMock = Enzyme.shallow(<Columns {...columnsMockProps} />)
    expect(columnsMock).toMatchSnapshot()
  })
})
