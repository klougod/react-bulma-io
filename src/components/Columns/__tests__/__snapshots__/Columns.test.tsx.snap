// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Columns component test should render without crash with props 1`] = `
<div
  className="columns columnsCLass"
/>
`;

exports[`Columns component test should render without crash without props 1`] = `
<div
  className="columns "
/>
`;
